integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,450 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+from dataclasses import asdict
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+from azure.core.credentials import AzureKeyCredential
+from azure.core.exceptions import ClientAuthenticationError, HttpResponseError, ResourceNotFoundError
+from azure.identity import DefaultAzureCredential
+from azure.search.documents import SearchClient
+from azure.search.documents.indexes import SearchIndexClient
+from azure.search.documents.indexes.models import (
+    HnswAlgorithmConfiguration,
+    HnswParameters,
+    SearchableField,
+    SearchField,
+    SearchFieldDataType,
+    SearchIndex,
+    SimpleField,
+    VectorSearch,
+    VectorSearchAlgorithmMetric,
+    VectorSearchProfile,
+)
+from azure.search.documents.models import VectorizedQuery
+from haystack import default_from_dict, default_to_dict
+from haystack.dataclasses import Document
+from haystack.document_stores.errors import DuplicateDocumentError
+from haystack.document_stores.types import DuplicatePolicy
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+from .errors import AzureAISearchDocumentStoreConfigError
+from .filters import normalize_filters
+
+type_mapping = {
+    str: "Edm.String",
+    bool: "Edm.Boolean",
+    int: "Edm.Int32",
+    float: "Edm.Double",
+    datetime: "Edm.DateTimeOffset",
+}
+
+MAX_UPLOAD_BATCH_SIZE = 1000
+
+DEFAULT_VECTOR_SEARCH = VectorSearch(
+    profiles=[
+        VectorSearchProfile(name="default-vector-config", algorithm_configuration_name="cosine-algorithm-config")
+    ],
+    algorithms=[
+        HnswAlgorithmConfiguration(
+            name="cosine-algorithm-config",
+            parameters=HnswParameters(
+                metric=VectorSearchAlgorithmMetric.COSINE,
+            ),
+        )
+    ],
+)
+
+logger = logging.getLogger(__name__)
+logging.getLogger("azure").setLevel(logging.ERROR)
+logging.getLogger("azure.identity").setLevel(logging.DEBUG)
+
+
+class AzureAISearchDocumentStore:
+    def __init__(
+        self,
+        *,
+        api_key: Secret = Secret.from_env_var("AZURE_SEARCH_API_KEY", strict=False),  # noqa: B008
+        azure_endpoint: Secret = Secret.from_env_var("AZURE_SEARCH_SERVICE_ENDPOINT", strict=False),  # noqa: B008
+        index_name: str = "default",
+        embedding_dimension: int = 768,
+        metadata_fields: Optional[Dict[str, type]] = None,
+        vector_search_configuration: VectorSearch = None,
+        create_index: bool = True,
```

Comment:
`create_index` is not documented in the docstring. Maybe it's not used and should be removed...